### ISSUE TITLE
persons_check_finished.php incorrectly detects names differing due to accents as the same name

### ISSUE THREAD

**Issue by jfly** [thewca/worldcubeassociation.org#1848]
See [this check for duplicates query](https://github.com/jfly/worldcubeassociation.org/blob/898142d37dd3b45b8c9c18ccc9b4ae996ada0ba6/webroot/results/admin/persons_check_finished.php#L245-L250).

Unfortunately, our version of mysql (5.6) does not support accent sensitive collations (they were added in mysql 8, see [this comment](https://github.com/thewca/worldcubeassociation.org/issues/258#issuecomment-323839095) for more details).

## `utf8mb4_unicode_ci` treats "Nguyễn" as equal to "Nguyen"

```mysql
MariaDB [(none)]> select convert('Nguyễn' using utf8mb4) = convert('Nguyen' using utf8mb4) collate utf8mb4_unicode_ci;
+------------------------------------------------------------------------------------------------+
| convert('Nguyễn' using utf8mb4) = convert('Nguyen' using utf8mb4) collate utf8mb4_unicode_ci   |
+------------------------------------------------------------------------------------------------+
|                                                                                              1 |
+------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

## `utf8mb4_unicode_520_ci` treats "Nguyễn" as equal to "Nguyen"

```mysql
MariaDB [(none)]> select convert('Nguyễn' using utf8mb4) = convert('Nguyen' using utf8mb4) collate utf8mb4_unicode_520_ci;
+----------------------------------------------------------------------------------------------------+
| convert('Nguyễn' using utf8mb4) = convert('Nguyen' using utf8mb4) collate utf8mb4_unicode_520_ci   |
+----------------------------------------------------------------------------------------------------+
|                                                                                                  1 |
+----------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

**Comment by AlbertoPdRF** [comment]
Should this issue be added to https://github.com/thewca/worldcubeassociation.org/projects/4?

**Comment by jfly** [comment]
Good call, added!

**Comment by SAuroux** [comment]
@tussosedan as we have upgraded to MySQL 8 since, could this issue now be resolved easily?

**Comment by tussosedan** [comment]
That's an interesting issue! MySQL 8 does have case sensitive and accent sensitive collations for the `utf8mb4` character set that we're using, while we currently use the case-insensitive collation `utf8mb4_unicode_ci`.

However, in the above and linked issues I see two different and possibly colliding needs -- on one hand, searching for `Lapicki` instead of `Łapicki` should perhaps find it, meaning the search should be insensitive to the difference. On the other hand, the duplicates search maybe should treat `Nguyễn` and `Nguyen` as different (although, shouldn't it be really recognized as a duplicate?).

On the technical side of things, we can `ALTER` specific columns to change their collations. This could be a heavy blocking operation that would require some downtime, and more importantly would need extensive testing to make sure nothing breaks unexpectedly afterwards. 

Another option is to convert collations on the fly inside specific queries, without changing the actual data in the tables themselves. This is easier for having a different behavior just for some scenarios, but could have performance issues on large data sets as it would prevent index seeks -- although it might be good enough, and/or overcome by defining and indexing virtual columns with different collations.

I think first of all we need to decide what is the correct behavior for each use case. Happy to hear more thoughts from @jonatanklosko, @viroulep and @jfly too.

**Comment by gregorbg** [comment]
@thewca/wrt Is this issue still relevant now that Alexandre's sanity-check project is a thing?

**Comment by SAuroux** [comment]
@gregorbg Yes, this is still relevant. We still use this script in today's result posting procedure and this issue is in fact entirely unrelated to the monthly sanity checks.

**Comment by dunkOnIT** [comment]
@gregorbg @danieljames-dj any idea if this is still relevant now that we've moved past PHP?
